client/cgo/verification.go

```diff
@@ -14,20 +14,25 @@ import (
 func main() {}
 
 //export cgoVerify
+// This cgoVerify returns 2 error codes,
+// the first one is the error of the response,
+// the second one is the error of the verification.
+// If the response error is in ErrorResponses,
+// the verification error will be -1.
 func cgoVerify(cType C.int,
 	cUname *C.char, cUnameSize C.int,
 	cKey unsafe.Pointer, cKeySize C.int,
 	cCurrentEpoch C.ulonglong,
 	cSavedSTR unsafe.Pointer, cStrSize C.int,
 	cPk unsafe.Pointer, cPkSize C.int,
-	cResponse *C.char, cResponseSize C.int) C.int {
+	cResponse *C.char, cResponseSize C.int) (C.int, C.int) {
 
 	if int(cUnameSize) == 0 ||
 		int(cKeySize) == 0 ||
 		(int(cStrSize) != sign.SignatureSize && int(cStrSize) != 0) ||
 		int(cPkSize) != sign.PublicKeySize ||
 		int(cResponseSize) == 0 {
-		return C.int(protocol.ErrorMalformedDirectoryMessage)
+		return C.int(protocol.ErrorMalformedDirectoryMessage), C.int(-1)
```

Comment:
Instead of returning `-1`, I'm thinking we should return an called something like `ErrorCouldNotVerify`, which is defined in `protocol.error`. Same goes for L47.